crates/precompile/Cargo.toml

```diff
@@ -48,11 +48,12 @@ ripemd.workspace = true
 aurora-engine-modexp.workspace = true
 
 # ecAdd, ecMul, ecPairing
-bn.workspace = true
+bn = { workspace = true, optional = true }
```

Comment:
Would for now, make bn not optional and remove the `bn` feature. This could potentially break if revm is used without default features.

`matter-labs-eip1962` if enabled would become the precompile that is included over bn.

Reply:
Let me think about how the defaults will work when switching between matter-labs-eip1962 and substrate-bn

Reply:
Yeah i would handle it differently from c-kzg, something similar to ecrecovery (k256 and secp256k1), k256 is not optional and if secp256k1 is not enabled it will be used.